ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS/LedgerPeers.hs

```diff
@@ -75,44 +78,69 @@ resolveLedgerPeers tracer
           Left (DNSError err) -> throwIO err
           Left (IOError  err) -> throwIO err
           Right resolver -> do
-            let lookups =
-                  [ (,) domain
-                      <$> withDNSSemaphore dnsSemaphore
-                            (dnsLookupWithTTL
-                              resolvConf
-                              resolver
-                              (dapDomain domain))
-                  | domain <- domains ]
+            let lookups' = snd $ foldl' f (rng0, []) domains
+                f (rng, lookups) domain =
+                  let (rng', rngResolve) =
+                        case domain of
+                          DomainAccessPoint {}    -> (rng, rng)
```

Comment:
hmm, if we ever need an `rng` in the domain lookup, then it will be quite easy to overlook it. It's better to pass a `Maybe rng` to the function (even if it becomes partial, but at least it fails explicitly).  But maybe that's not possible.

Reply:
If we choose to stick with the current solution, we should include a comment in the lookup function that the `rng` should not be used in plain domain lookups.

Reply:
Also another alternative is to always split it, but I've avoided it to speed things up. Maybe that won't be an issue?